### PR TITLE
AB#31 Add Project's Profile

### PR DESCRIPTION
**Pull request by paola-quintanilla-jalafund**
image

### CODE REVIEW

File: project-api/src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        public GetProject(IProjectsService projectsService)
+        {
+            this.projectsService = projectsService;
+        }
+
+        // [FunctionName("GetUser")]
+        // [OpenApiOperation(operationId: "Get")]
+        // [OpenApiParameter(name: "name", In = ParameterLocation.Query, Required = true, Type = typeof(string), Description = "The **Name** parameter")]
+        // [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "text/plain", bodyType: typeof(List<string>), Description = "The OK response")]
```
Comment (cristian-chavez-jalafund) [inline project-api/src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]: please remove the unnecessary comments

**Review by cristian-chavez-jalafund — CHANGES_REQUESTED** [review]

File: project-api/src/core/Jalasoft.TeamUp.Projects.Core.Interfaces/IProjectsService.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Jalasoft.TeamUp.Projects.Core.Interfaces
+{
+    using System;
+    using Jalasoft.TeamUp.Projects.Models;
+
+    public interface IProjectsService
+    {
+        Project GetProject(Guid id);
```
Comment (Raul-Gamarra-jalafund) [inline project-api/src/core/Jalasoft.TeamUp.Projects.Core.Interfaces/IProjectsService.cs]: please add access modifier to this method

File: project-api/src/dal/Jalasoft.TeamUp.Projects.DAL.Interfaces/IProjectsRepository.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Jalasoft.TeamUp.Projects.DAL.Interfaces
+{
+    using System;
+    using Jalasoft.TeamUp.Projects.Models;
+
+    public interface IProjectsRepository
+    {
+        Project GetProject(Guid id);
```
Comment (Raul-Gamarra-jalafund) [inline project-api/src/dal/Jalasoft.TeamUp.Projects.DAL.Interfaces/IProjectsRepository.cs]: please add access modifier

**Review by Raul-Gamarra-jalafund — DISMISSED** [review]

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetHealth.cs [hunk 4]
```diff
@@ -16,21 +16,20 @@
 
     public class GetHealth
     {
-        private readonly IHealthsService projectsService;
+        private readonly IHealthsService healthsService;
 
-        public GetHealth(IHealthsService projectsService)
+        public GetHealth(IHealthsService healthsService)
```
Comment (martin-callejas-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetHealth.cs]: I know it's not your code but what's the convention for the health endpoint here? Can you check and align that for me please? It's different from resumes project atm

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetHealth.cs [hunk 5]
```diff
@@ -16,21 +16,20 @@
...
         {
-            this.projectsService = projectsService;
+            this.healthsService = healthsService;
```
Comment (martin-callejas-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetHealth.cs]: health**s** doesn't sound ok to me, please double check

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    public class GetProject
+    {
+        private readonly IProjectsService projectsService;
+
+        public GetProject(IProjectsService projectsService)
+        {
+            this.projectsService = projectsService;
+        }
+
+        [FunctionName("GetProject")]
```
Comment (martin-callejas-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]: As it stands right now, this is not a RESTful API. Let's adhere to resources naming good practices, ok? Your endpoint should be something like `GET api/projects/{id}` :) 
Read more here: https://restfulapi.net/resource-naming/

Big refactor incoming...

File: src/api/Jalasoft.TeamUp.Projects.API/Properties/launchSettings.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (martin-callejas-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Properties/launchSettings.json]: I think we don't want to commit this file?

File: src/api/Jalasoft.TeamUp.Projects.API/local.settings.json [hunk 8]
```diff
@@ -1,7 +1,11 @@
...
-    "Values": {
-        "AzureWebJobsStorage": "UseDevelopmentStorage=true",
-        "FUNCTIONS_WORKER_RUNTIME": "dotnet"
-    }
+  "IsEncrypted": false,
+  "Values": {
+    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
+    "FUNCTIONS_WORKER_RUNTIME": "dotnet"
+  },
+  "Host": {
```
Comment (martin-callejas-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/local.settings.json:7]: is this a common agreement with the team? it's intentional? do Resumes does the same CORS setting?

File: src/core/Jalasoft.TeamUp.Projects.Core/HealthsService.cs [hunk 9]
```diff
@@ -9,16 +9,16 @@
 
     public class HealthsService : IHealthsService
```
Comment (martin-callejas-jalafund) [inline src/core/Jalasoft.TeamUp.Projects.Core/HealthsService.cs]: singular for health, please. And if you're gonna rename trough VS, CTRL + R + R it's your friend https://docs.microsoft.com/en-us/visualstudio/ide/reference/rename?view=vs-2019

File: src/core/Jalasoft.TeamUp.Projects.Core/HealthsService.cs [hunk 10]
```diff
@@ -9,16 +9,16 @@
...
     {
-        private readonly IHealthsRepository projectsRepository;
+        private readonly IHealthsRepository healthsRepository;
```
Comment (martin-callejas-jalafund) [inline src/core/Jalasoft.TeamUp.Projects.Core/HealthsService.cs]: singular... everywhere

File: src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsRepository.cs [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Jalasoft.TeamUp.Projects.DAL.Interfaces;
+    using Jalasoft.TeamUp.Projects.Models;
+
+    public class ProjectsRepository : IProjectsRepository
+    {
+        private static IEnumerable<Project> projects = new List<Project>() { new Project() { Id = Guid.Parse("5a7939fd-59de-44bd-a092-f5d8434584de"), Name = "TeamUp", Contact = new Contact() { Name = "Jose Ecos", IdResume = Guid.Parse("5a7939fd-59de-44bd-a092-f5d8434584de") }, Description = "Centralize resumes and project", Logo = "https://www.example.com/images/dinosaur.jpg", MemberList = new List<Contact>(), State = true, TextInvitation = "You are invited to be part of TeamUp" } };
```
Comment (martin-callejas-jalafund) [inline src/dal/Jalasoft.TeamUp.Projects.DAL/ProjectsRepository.cs]: can we improve formatting here? up to you

**Review by martin-callejas-jalafund — DISMISSED** [review]
Good job! Would need more feedback from the backend trainers though

File: src/api/Jalasoft.TeamUp.Projects.API/local.settings.json (hunk 8 above)
Comment (paola-quintanilla-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/local.settings.json:7]: In the backend team yes, we had a common agreement

File: src/api/Jalasoft.TeamUp.Projects.API/Properties/launchSettings.json (hunk 7 above)
Comment (paola-quintanilla-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Properties/launchSettings.json]: I think It is necessary for launch in port 7072

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs (hunk 6 above)
Comment (paola-quintanilla-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]: the endpoint to getProject is something like that:
Get api/project/{id}

image

Comment (paola-quintanilla-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]:
```suggestion
       [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route ="projects/{id:guid}")] HttpRequest req)
```

Comment (martin-callejas-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]: @Raul-Gamarra-jalafund here man

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetHealth.cs (hunk 4 above)
Comment (paola-quintanilla-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetHealth.cs]: I could read something about that, and health is used without **s** :D

**Review by cristian-chavez-jalafund — APPROVED** [review]

**Review by cristian-chavez-jalafund — APPROVED** [review]

**Review by rodrigo-baldiviezo-jalafund — APPROVED** [review]

**Review by ela-ocana-jalafund — APPROVED** [review]

**Review by lina-ramirez-jalafund — APPROVED** [review]

**Pull request merged**